### PR TITLE
[microTVM] Add Nucleo stm32l4r5zi board to zephyr

### PR DESCRIPTION
**Pull request by mehrdadh**
This PR:
- adds stm32l4r5zi as another target to micro targets
- adds stm32l4r5zi config files for zephyr test and zephyr AOT test

cc @areusch @gromero

### CODE REVIEW

**Comment by mehrdadh** [comment]
`test_zephyr` and `test_zephyr_aot` has passed on stm32l4r5zi board on my local machine.

File: python/tvm/micro/contrib/zephyr.py [hunk 1]
```diff
@@ -408,6 +408,7 @@ def _get_nrf_device_args(self):
     BOARD_USB_FIND_KW = {
         "nucleo_f746zg": {"idVendor": 0x0483, "idProduct": 0x374B},
         "stm32f746g_disco": {"idVendor": 0x0483, "idProduct": 0x374B},
+        "nucleo_l4r5zi": {"idVendor": 0x0483, "idProduct": 0x374B},
```
Comment (leandron) [inline python/tvm/micro/contrib/zephyr.py:411]:
```suggestion
        "nucleo_l4r5zi": {"idVendor": 0x0483, "idProduct": 0x374B},
        "nucleo_f746zg": {"idVendor": 0x0483, "idProduct": 0x374B},
        "stm32f746g_disco": {"idVendor": 0x0483, "idProduct": 0x374B},
```

This is a massive nit, but I suggest we keep the boards, for our own sanity, alphabetically here, where possible.

File: python/tvm/target/target.py [hunk 2]
```diff
@@ -284,6 +284,7 @@ def intel_graphics(model="unknown", options=None):
     "stm32f746xx": ["-mcpu=cortex-m7", "-march=armv7e-m"],
     "nrf5340dk": ["-mcpu=cortex-m33"],
     "mps2_an521": ["-mcpu=cortex-m33"],
+    "stm32l4r5zi": ["-mcpu=cortex-m4"],
```
Comment (leandron) [inline python/tvm/target/target.py:287]:
```suggestion
    "mps2_an521": ["-mcpu=cortex-m33"],
    "nrf5340dk": ["-mcpu=cortex-m33"],
    "stm32f746xx": ["-mcpu=cortex-m7", "-march=armv7e-m"],
    "stm32l4r5zi": ["-mcpu=cortex-m4"],
```

Again, this is a massive nit, but I suggest we keep the boards, for our own sanity, alphabetically here, where possible.

File: tests/micro/zephyr/conftest.py [hunk 3]
```diff
@@ -27,6 +27,7 @@
     "stm32f746xx_nucleo": ("stm32f746xx", "nucleo_f746zg"),
     "stm32f746xx_disco": ("stm32f746xx", "stm32f746g_disco"),
     "nrf5340dk": ("nrf5340dk", "nrf5340dk_nrf5340_cpuapp"),
+    "stm32l4r5zi_nucleo": ("stm32l4r5zi", "nucleo_l4r5zi"),
     "mps2_an521": ("mps2_an521", "mps2_an521-qemu"),
```
Comment (leandron) [inline tests/micro/zephyr/conftest.py:31]:
```suggestion
    "mps2_an521": ("mps2_an521", "mps2_an521-qemu"),
    "nrf5340dk": ("nrf5340dk", "nrf5340dk_nrf5340_cpuapp"),
    "stm32f746xx_disco": ("stm32f746xx", "stm32f746g_disco"),
    "stm32f746xx_nucleo": ("stm32f746xx", "nucleo_f746zg"),
    "stm32l4r5zi_nucleo": ("stm32l4r5zi", "nucleo_l4r5zi"),
```

**Review by leandron — COMMENTED** [review]
LGTM overall. Just a few nits, trying to keep the code organised.

File: python/tvm/micro/contrib/zephyr.py (hunk 1 above)
Comment (areusch) [reply python/tvm/micro/contrib/zephyr.py:411]: i agree with alphabetic sorting

File: tests/micro/zephyr/test_zephyr_aot.py [hunk 4]
```diff
@@ -214,6 +214,9 @@ def test_tflite(platform, west_cmd, skip_build, tvm_debug):
 
 
 def test_qemu_make_fail(platform, west_cmd, skip_build, tvm_debug):
+    if platform not in ["host", "mps2_an521"]:
```
Comment (areusch) [inline tests/micro/zephyr/test_zephyr_aot.py:217]: should we add a decorator to tvm.testing?

**Comment by mehrdadh** [comment]
@leandron thanks for the review. I changed the order.

File: tests/micro/zephyr/test_zephyr_aot.py (hunk 4 above)
Comment (mehrdadh) [reply tests/micro/zephyr/test_zephyr_aot.py]: There are more changes that need to be done here. I removed this changes and will send another PR.

**Review by leandron — APPROVED** [review]
LGTM, thanks @mehrdadh

**Pull request merged**